bluesky/callbacks/kafka.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+
+from kafka import KafkaProducer, KafkaConsumer
+from bluesky.run_engine import Dispatcher
+import pickle
+
+from bluesky.callbacks.core import CallbackBase
+
+from ..run_engine import Dispatcher, DocumentNames
+
+import time
+
+
+class Producer(CallbackBase):
+    """
+    A callback that publishes documents to kafka.
+
+    Parameters
+    ----------
+    bootstrap_servers: list
+        list of servers the brokers run on
+    topic: str
+        topic to push to
+
+    Example
+    -------
+    >>> producer = Producer(bootstrap_servers=['localhost:9092'], topic='analysis')
+    """
+    def __init__(self, bootstrap_servers, topic):
+        self._publisher = KafkaProducer(bootstrap_servers=bootstrap_servers)
+        self._topic = topic
+
+    def __call__(self, name, doc):
+        message = b' '.join([name.encode(), pickle.dumps(doc)])
```

Comment:
How tied are we to pickle here?  Short term this is probably ok, but mid to long term it would be better to use an encoding that can be read from non-python client.